src/JWT/Builder/JwtBuilder.cs

```diff
@@ -251,6 +248,22 @@ public string Encode()
             return _encoder.Encode(_jwt.Header, _jwt.Payload, _secrets?[0]);
         }
 
+        public string Encode<T>(T payload)
+        {
+            EnsureCanEncode();
+
+            var payloadAsDictionary = payload.GetType()

```

Comment:
Not sure if this will work. You basically doing a native serialization. What if the value is not a primitive type? I'd not add it, at least not in this PR.